### PR TITLE
Samples view for data discovery project

### PR DESCRIPTION
**Pull request by tfrcarvalho**
* New samples view
* Uses new `InfiniteTable` using infinite loader with loading animation
* Column selector
* Guarded by feature flag
* New style for checkboxes
* New plus icon

* TODO on samples view: selectable rows, sort on infinite table

samples_view

### CODE REVIEW

File: app/assets/src/components/visualizations/table/BaseTable.jsx [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+import MultipleDropdown from "~ui/controls/dropdowns/MultipleDropdown";
+import PlusIcon from "~ui/icons/PlusIcon";
+import { find, map } from "lodash";
+import cs from "./base_table.scss";
+import cx from "classnames";
+
+class BaseTable extends React.Component {
+  // This class is a wrapper class to React Virtualized Table.
+  // Sets some default values and style to guarantee consistency of
+  // tables accross the site.s
```
Comment (jshoe) [inline app/assets/src/components/visualizations/table/BaseTable.jsx]: "site.s"

File: app/controllers/samples_controller.rb [hunk 2]
```diff
@@ -82,7 +83,7 @@ def index
     # Query by name for a Sample attribute or pathogen name in the Sample.
     if name_search_query.present?
       # Pass in a scope of pipeline runs using current_power
-      pipeline_run_ids = current_power.pipeline_runs.top_completed_runs.pluck(:id)
+      pipeline_run_ids = current_fpower.pipeline_runs.top_completed_runs.pluck(:id)
```
Comment (jshoe) [inline app/controllers/samples_controller.rb]: "current_fpower"

File: app/controllers/samples_controller.rb [hunk 3]
```diff
@@ -121,7 +122,10 @@ def index
   def index_v2
     only_library = ActiveModel::Type::Boolean.new.cast(params[:onlyLibrary])
     exclude_library = ActiveModel::Type::Boolean.new.cast(params[:excludeLibrary])
+    limit = params[:limit] ? params[:limit].to_i : MAX_PAGE_SIZE_V2
+    offset = params[:offset].to_i
 
+    Rails.logger.debug("params=#{params}")
```
Comment (jshoe) [inline app/controllers/samples_controller.rb]: remove?

File: app/assets/src/components/visualizations/table/BaseTable.jsx (hunk 1 above)
Comment (tfrcarvalho) [reply app/assets/src/components/visualizations/table/BaseTable.jsx]: Fixed

File: app/controllers/samples_controller.rb (hunk 3 above)
Comment (tfrcarvalho) [reply app/controllers/samples_controller.rb]: Fixed

File: app/controllers/samples_controller.rb (hunk 2 above)
Comment (tfrcarvalho) [reply app/controllers/samples_controller.rb]: Fixed

**Pull request merged**